lib/services/matchers.go

```diff
@@ -123,9 +125,17 @@ type ResourceSeenKey struct{ name, addr string }
 // it filters out the non-matched clusters on the kube service and the kube service
 // is modified in place with only the matched clusters. Deduplication for resource `KubeService`
 // is not provided but is provided for kind `KubernetesCluster`.
-func MatchResourceByFilters(resource types.ResourceWithLabels, filter MatchResourceFilter, seenMap map[ResourceSeenKey]struct{}) (bool, error) {
+func MatchResourceByFilters(
```

Comment:
This matcher code is used by the RBAC engine. I don't think you want to touch how matching works. This task is really just to avoid showing the labels in the UI. I think you can solve this client side without changing Teleport.